src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -0,0 +1,671 @@
+				validatedMedicalInventories = medicalInventoryManager.getMedicalInventoryByStatusAndInventoryType(validated, inventoryType);
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (draftMedicalInventories.isEmpty() && validatedMedicalInventories.isEmpty()) {
+				InventoryEdit inventoryEdit = new InventoryEdit();
+				InventoryEdit.addInventoryListener(InventoryBrowser.this);
+				inventoryEdit.showAsModal(InventoryBrowser.this);
+			} else {
+				MessageDialog.error(null, "angal.inventory.cannotcreateanotherinventorywithstatusdraft.msg");
+				return;
+			}
+		});
+		return jButtonNew;
+	}
+
+	private JButton getEditButton() {
+		jButtonEdit = new JButton(MessageBundle.getMessage("angal.common.edit.btn"));
+		jButtonEdit.setMnemonic(MessageBundle.getMnemonic("angal.common.edit.btn.key"));
+		jButtonEdit.setEnabled(false);
+		jButtonEdit.addActionListener(actionEvent -> {
+			MedicalInventory inventory = new MedicalInventory();
+			if (jTableInventory.getSelectedRowCount() > 1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectonlyoneinventory.msg");
+				return;
+			}
+			int selectedRow = jTableInventory.getSelectedRow();
+			if (selectedRow == -1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectinventory.msg");
+				return;
+			}
+			inventory = inventoryList.get(selectedRow);
+			if (inventory.getStatus().equals(InventoryStatus.canceled.toString())) {
+				MessageDialog.error(null, "angal.inventory.cancelednoteditable.msg");
+				return;
+			}
+
+			if (inventory.getStatus().equals(InventoryStatus.done.toString())) {
+				MessageDialog.error(null, "angal.inventory.donenoteditable.msg");
+				return;
+			}
+			InventoryEdit inventoryEdit = new InventoryEdit(inventory, "update");
+			InventoryEdit.addInventoryListener(InventoryBrowser.this);
+			inventoryEdit.showAsModal(InventoryBrowser.this);
+		});
+		return jButtonEdit;
+	}
+
+	private JButton getViewButton() {
+		jButtonView = new JButton(MessageBundle.getMessage("angal.inventory.view.btn"));
+		jButtonView.setMnemonic(MessageBundle.getMnemonic("angal.inventory.view.btn.key"));
+		jButtonView.setEnabled(false);
+		jButtonView.addActionListener(actionEvent -> {
+			MedicalInventory inventory = new MedicalInventory();
+			if (jTableInventory.getSelectedRowCount() > 1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectonlyoneinventory.msg");
+				return;
+			}
+			int selectedRow = jTableInventory.getSelectedRow();
+			if (selectedRow == -1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectinventory.msg");
+				return;
+			}
+			if (selectedRow > -1) {
+				inventory = inventoryList.get(selectedRow);
+				InventoryEdit inventoryEdit = new InventoryEdit(inventory, "view");
+				InventoryEdit.addInventoryListener(InventoryBrowser.this);
+				inventoryEdit.showAsModal(InventoryBrowser.this);
```

Comment:
```suggestion
				inventoryEdit.showAsModal(this);
```
unnecessary qualifier in this context